Bug: reinhardt-core still pulls in tokio/mio unconditionally, breaking WASM builds

## Summary

Despite the fix in PR #111 (resolving #81), `reinhardt-core` still has unconditional `hyper` and `tokio` dependencies that are not gated behind `cfg(not(target_arch = "wasm32"))`. This causes WASM builds (`wasm32-unknown-unknown`) to fail with `mio` compilation errors.

## Environment

- Rust: nightly-2025-04-15 (edition 2024)
- Target: `wasm32-unknown-unknown`
- reinhardt-web: v0.1.0-rc.15 (commit 7790c5ae)
- OS: macOS (Darwin 25.3.0)

## Steps to Reproduce

1. Create a project that depends on `reinhardt-web` (which re-exports `reinhardt-core`, `reinhardt-pages`)
2. Configure `crate-type = ["cdylib", "rlib"]` for the lib target
3. Run `cargo build --lib --target wasm32-unknown-unknown`

## Expected Behavior

The WASM build should succeed — `reinhardt-pages` and `reinhardt-core` should gate server-only dependencies behind `cfg(not(target_arch = "wasm32"))`.

## Actual Behavior

Build fails with 49 errors in `mio`:

```
error[E0432]: unresolved import `crate::sys`
   --> mio-1.1.1/src/io_source.rs:14:5
error[E0599]: no method named `deregister` found for struct `IoSource`
   --> mio-1.1.1/src/io_source.rs:96:14
error: could not compile `mio` (lib) due to 49 previous errors
```

## Root Cause

Dependency chain verified via `cargo tree --target wasm32-unknown-unknown -i mio`:

```
mio v1.1.1
└── tokio v1.50.0
    ├── h2 v0.4.13
    │   └── hyper v1.8.1
    │       └── reinhardt-core v0.1.0-rc.15
    │           └── reinhardt-pages-macros v0.1.0-rc.15 (proc-macro)
    │               └── reinhardt-pages v0.1.0-rc.15
    │                   └── reinhardt-web v0.1.0-rc.15
```

PR #111 gated `tokio`/`hyper` in `reinhardt-core` but the fix appears to be incomplete or has regressed. The current `reinhardt-core` Cargo.toml still exposes `hyper` (and transitively `tokio` → `mio`) to WASM builds.

## Proposed Fix

Move remaining server-only dependencies in `reinhardt-core/Cargo.toml` behind target cfg:

```toml
[target.'cfg(not(target_arch = "wasm32"))'.dependencies]
tokio = { workspace = true, features = ["rt", "sync", "time"] }
hyper = { workspace = true }
h2 = { workspace = true }
```

## Related Issues

- #81 — Original report (closed, but incomplete fix)
- PR #111 — Partial fix that gated some, but not all, server-side deps

## Impact

**Critical** — WASM frontend builds are completely broken. Projects using `reinhardt-pages` for client-side SPA cannot compile.

--- a/crates/reinhardt-pages/macros/Cargo.toml
+++ b/crates/reinhardt-pages/macros/Cargo.toml
@@ -20,9 +20,6 @@ default = []
 # DSL definitions (new canonical location)
 reinhardt-manouche = { workspace = true }
 
-# Security utilities for URL validation (Fixes #849)
-reinhardt-core = { workspace = true }
-
 # Procedural macro essentials
 syn = { workspace = true, features = ["full", "extra-traits"] }
 quote = { workspace = true }

--- a/crates/reinhardt-pages/macros/src/page/validator.rs
+++ b/crates/reinhardt-pages/macros/src/page/validator.rs
@@ -14,12 +14,32 @@
 use proc_macro2::Span;
 use syn::{Expr, Result};
 
-use reinhardt_core::security::xss::is_safe_url;
 use reinhardt_manouche::core::{
 	PageAttr, PageBody, PageComponent, PageElement, PageElse, PageEvent, PageMacro, PageNode,
 	PageWatch, TypedPageAttr, TypedPageBody, TypedPageComponent, TypedPageElement, TypedPageElse,
 	TypedPageFor, TypedPageIf, TypedPageMacro, TypedPageNode, TypedPageWatch, types::AttrValue,
 };
+
+/// Check if a URL is safe (no dangerous schemes like javascript:).
+///
+/// Inlined from `reinhardt_core::security::xss::is_safe_url` to avoid
+/// pulling the full reinhardt-core dependency chain (hyper/tokio/mio)
+/// into this proc-macro crate, which breaks WASM builds. (Fixes #3226)
+fn is_safe_url(url: &str) -> bool {
+	let url_lower = url.to_lowercase();
+
+	// Allow relative URLs and anchor links (but NOT parent traversal)
+	if url.starts_with('/') || url.starts_with("./") || url.starts_with('#') {
+		return true;
+	}
+
+	// Allow only safe protocols
+	let safe_protocols = ["http://", "https://", "mailto:", "ftp://", "ftps://"];
+
+	safe_protocols
+		.iter()
+		.any(|protocol| url_lower.starts_with(protocol))
+}
 
 /// Validates and transforms the entire PageMacro AST into a typed AST.
 ///
@@ -818,6 +838,7 @@ fn validate_element_nesting(elem: &PageElement, parent_tags: &[String]) -> Resul
 #[cfg(test)]
 mod tests {
 	use super::*;
+	use rstest::rstest;
 	use syn::parse_quote;
 
 	#[test]
@@ -1422,5 +1443,22 @@ mod tests {
 		let result =
 			validate_button_accessibility(&attrs, &children, proc_macro2::Span::call_site());
 		assert!(result.is_ok());
+	}
+
+	#[rstest]
+	#[case("https://example.com", true)]
+	#[case("http://example.com", true)]
+	#[case("mailto:user@example.com", true)]
+	#[case("ftp://files.example.com", true)]
+	#[case("ftps://files.example.com", true)]
+	#[case("/relative/path", true)]
+	#[case("./local/path", true)]
+	#[case("#anchor", true)]
+	#[case("javascript:alert(1)", false)]
+	#[case("data:text/html,<script>alert(1)</script>", false)]
+	#[case("vbscript:msgbox", false)]
+	#[case("JAVASCRIPT:alert(1)", false)]
+	fn test_is_safe_url(#[case] url: &str, #[case] expected: bool) {
+		assert_eq!(is_safe_url(url), expected);
 	}
 }

